### PR TITLE
M2P-82 Read bolt cart from sections instead of AJAX call

### PR DESCRIPTION
**Pull request by vitaliyreznikov**
In the next PR we save old code for onestep case and do some refactoring for multistep case

Fixes: [M2P-82]

#changelog Plugin refactoring

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Refactoring

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Comment by mccottry** [comment]
@vitaliyreznikov 
On its own, this PR doesn't make sense.  It is certainly familiar reused code, but here we see it pulled out of nowhere, supposedly moved with a purpose, but that is not clear.

In this case, you should do as @danac-gs has suggested and place this code alongside the other relevant code so that one can see a clear picture into why this is being done and how effectively it achieves its intent.

**Comment by vitaliyreznikov** [comment]
Yes, Leon, sorry for confusing. I just duplicated code to refactor multistep case in next PR. We can't use section module for onestep case and probably will remove support for one step soon.

**Comment by mccottry** [comment]
so, @vitaliyreznikov, should we just merge the next PR to this and review those together?

File: view/frontend/templates/js/replacejs.phtml [hunk 1]
```diff
@@ -684,81 +687,130 @@ ob_start();
...
-            var config = boltConfig.always_present_checkout && (numItems && numItems !=="0" || dataChanged)? {floatingButtonMode: dataChanged ? "show_cart" : "show_cart_on_hover" } : {};
-            var BC = BoltCheckout.configure(cart, hintBarrier.promise, callbacks, config);
-        };
+            if (!wasConfigureCalled) {
+                // temporary disable always present checkout button for refactoring reason
+                // will remove this line in the next PR
+                var config = {};
+                BC = BoltCheckout.configure(cartBarrier.promise, hintBarrier.promise, callbacks, config);
+                wasConfigureCalled = true;
+            }
```
[the remark this answers is not in the document]
Comment (vitaliyreznikov) [reply view/frontend/templates/js/replacejs.phtml:811]: Sounds good Leon. Let me know if I can help with this fix.
Do you think we can go ahead, merge all stuff and fix the reward issue in separate PR?

[the remark this answers is not in the document]
Comment (mccottry) [reply view/frontend/templates/js/replacejs.phtml:811]: Good Idea!  @danac-gs, can we get this approved, please?

**Comment by vitaliyreznikov** [comment]
Merging this because it was approved by Leon and Kazuki

**Pull request merged**